src/drivers/distance_sensor/ll40ls/ll40ls.cpp

```diff
@@ -89,159 +63,227 @@ namespace ll40ls
+start_bus(const int bus, const uint8_t rotation)
 {
-	instance = new LidarLiteI2C(i2c_bus.busnum, rotation);
+	if (instance != nullptr) {
+		PX4_ERR("already started");
+		return PX4_OK;
+	}
+
+	// Instantiate the driver.
+	instance = new LidarLiteI2C(bus, rotation);
+
+	if (instance == nullptr) {
+		PX4_ERR("Failed to instantiate the driver");
+		delete instance;
+		return PX4_ERROR;
+	}
 
-	if (instance->init() != OK) {
-		stop();
-		PX4_INFO("LidarLiteI2C - no device on bus %u", (unsigned)i2c_bus.busid);
+	// Initialize the sensor.
+	if (instance->init() != PX4_OK) {
+		PX4_ERR("Failed to initialize LidarLite on bus = %u", bus);
+		delete instance;
+		instance = nullptr;
 		return PX4_ERROR;
 	}
 
+	// Start the driver.
+	instance->start();
+
+	PX4_INFO("driver started");
 	return PX4_OK;
 }
 
 /**
- * @brief Stops the driver
+ * Start the pwm driver.
+ *
+ * This function only returns if the sensor is up and running
+ * or could not be detected successfully.
  */
-void stop()
+int
+start_pwm(const uint8_t rotation)
 {
 	if (instance != nullptr) {
+		PX4_ERR("already started");
+		return PX4_OK;
+	}
+
+	instance = new LidarLitePWM(rotation);
+
+	if (instance == nullptr) {
+		PX4_ERR("Failed to instantiate the driver");
 		delete instance;
-		instance = nullptr;
+		return PX4_ERROR;
+	}
 
-	} else {
-		PX4_ERR("driver not running");
+	// Initialize the sensor.
+	if (instance->init() != PX4_OK) {
+		PX4_ERR("Failed to initialize LidarLite pwm.");
+		delete instance;
+		instance = nullptr;
+		return PX4_ERROR;
 	}
 
+	// Start the driver.
+	instance->start();
 
+	PX4_INFO("driver started");
+	return PX4_OK;
 }
 
 /**
  * @brief Prints status info about the driver.
  */
-void
-info()
+int
+status()
 {
-	if (!instance) {
-		warnx("No ll40ls driver running");
-		return;
+	if (instance == nullptr) {
+		PX4_ERR("driver not running");
+		return PX4_ERROR;
 	}
 
-	printf("state @ %p\n", instance);
 	instance->print_info();
+	return PX4_OK;
 }
 
 /**
- * @brief Dumps the register information.
+ * @brief Stops the driver
  */
-void
-regdump()
+int
+stop()
 {
-	if (!instance) {
-		warnx("No ll40ls driver running");
-		return;
+	if (instance != nullptr) {
+		delete instance;
+		instance = nullptr;
 	}
 
-	printf("regdump @ %p\n", instance);
-	instance->print_registers();
+	PX4_INFO("driver stopped");
+	return PX4_OK;
 }
 
 /**
  * @brief Displays driver usage at the console.
  */
-void
+int
 usage()
 {
-	PX4_INFO("missing command: try 'start', 'stop', 'info', 'info' or 'regdump' [i2c|pwm]");
-	PX4_INFO("options for I2C:");
-	PX4_INFO("    -X only external bus");
-#ifdef PX4_I2C_BUS_ONBOARD
-	PX4_INFO("    -I only internal bus");
-#endif
-	PX4_INFO("E.g. ll40ls start i2c -R 0");
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+
+I2C bus driver for LidarLite rangefinders.
+
+The sensor/driver must be enabled using the parameter SENS_EN_LL40LS.
+
+Setup/usage information: https://docs.px4.io/v1.9.0/en/sensor/lidar_lite.htmls
```

Comment:
I would probably not reference the v1.9.0 version here but just use `master`.

Reply:
Nice catch! updated. Thanks!